helmclient.go

```diff
@@ -394,6 +395,46 @@ func (c *Client) InstallFromTarball(path, ns string, options ...helmclient.Insta
 	return nil
 }
 
+// RunReleaseTest runs the tests for a Helm Release. The releaseName is the name of the
+// Helm Release that is set when the Helm Chart is installed.
+func (c *Client) RunReleaseTest(releaseName string, options ...helmclient.ReleaseTestOption) error {
+	c.logger.Log("level", "debug", "message", fmt.Sprintf("running tests for release '%s'", releaseName))
+
+	t, err := c.newTunnel()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	defer c.closeTunnel(t)
+
+	resChan, errChan := c.newHelmClientFromTunnel(t).RunReleaseTest(releaseName, helmclient.ReleaseTestTimeout(300))
+	if IsReleaseNotFound(err) {
+		return backoff.Permanent(microerror.Maskf(releaseNotFoundError, releaseName))
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for {
```

Comment:
Why do we need the `for` here? In case none of the channels returns something yet?

Reply:
Helm sends multiple results to the resChan. The for loop is so we capture all of them. The time.After timeout is to make sure we don't get stuck in the loop.